python/hail/expr/aggregators.py

```diff
@@ -56,7 +56,7 @@ def _agg_func(name, aggregable, ret_type, *args):
 
 def _check_agg_bindings(expr):
     bound_references = {ast.name for ast in expr._ast.search(lambda ast: isinstance(ast, VariableReference))}
-    bind_vars = {ast.uid for ast in expr._ast.search(lambda ast: isinstance(ast, Bind))}
+    bind_vars = {uid for ast in expr._ast.search(lambda ast: isinstance(ast, Bind)) for uid in ast.uids}
```

Comment:
was this avoiding tests? can we make sure it's covered?

Reply:
this was getting picked up by the mendel_errors test in #3431. I'll try to figure out why it wasn't before.

Reply:
ok, I added a test to test_aggregate that uses `hl.bind` inside an aggregator. it catches the error that's thrown before this gets fixed.

Reply:
fun. i just hit this error. pulling these changes now :-D